docs/src/lang/apalache-operators.md

```diff
@@ -43,6 +43,53 @@ True
 >> a = (b == "c") # a' := (b = "c")
 ```
 
+----------------------------------------------------------------------------
+
+<a name="Gen"></a>
+## Value generators
+
+**Notation:** `Gen(bound)`
+
+**LaTeX notation:** `Gen(bound)`
+
+**Arguments:** One argument: an integer literal or a constant expression (of
+the integer type).
+
+**Apalache type:** `Int => a`, for some type `a`.
+
+**Effect:** A generator of a data structure. Given a positive integer `bound`,
+and assuming that the type of the operator application is known, we recursively
+generate a TLA+ data structure as a tree, whose width is bound by the number
+`bound`.
+
+**Determinism:** The generated data structure is unrestricted.  It is
+effectively implementing data non-determinism.
+
+**Errors:**
+If the type of `Gen` cannot be inferred from its application context,
+or if `bound` is not an integers, Apalache reports an error.
+
+**Example in TLA+:**
+
+```tla
+\* produce an unrestricted integer
+LET \* @type: Int;
+    Gen(1)
+IN
+\* produce a set of integers up to 10 elements
+LET \* @type: Set(Int);
+    Gen(10)
+IN
+\* produce a sequence of up to 10 elements
+\* that are integers up to 10 elements each
+LET \* @type: Seq(Set(Int));
+    Gen(10)
+IN
+...
```

Comment:
Yes, these are pure data structures. If you want to restrict them, you can write TLA+ formulas.

Reply:
These are pure data structures, but to what end? Wouldn't any actual usage of them bind the generated values to some local variable that you can then test/inspect in some way?

Reply:
E.g., as we see here: https://github.com/informalsystems/apalache/blob/86a2f04ff9b60275e97f8c1fa56b87e816c93536/test/tla/TestGen.tla#L16-L17

Reply:
So what I'm suggesting is that if this is the actual way these get used, and we are here introducing the operators, it would probably be clearer to present them in a context/situation?

Another thing that is throwing me off here, is that I wasn't aware that TLA allows for expressions that are not variable bindings in the left side of a `LET/IN`, and I'm not even sure what the semantics of that is. Is it just a discarded value? I'm not the biggest TLA expert of course, but I kind of suspect that if these two things throw me off, they are also likely to confuse other readers.

Reply:
Oh, I see! I just forgot to bind them to names. Totally missed that.